modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,724 @@
+    table["Balance"].caption = value
+end)
+
+
+local function market(event)
+    local player = game.players[event.player_index]
+    Store.set(balances,player,player.get_item_count("coin"))
+end
+
+local function insideBox(box, pos)
+    local x1 = box[1][1]
+    local y1 = box[1][2]
+    local x2 = box[2][1]
+    local y2 = box[2][2]
+
+    local px = pos.x
+    local py = pos.y
+    return px >= x1 and px <= x2 and py >= y1 and py <= y2
+end
+
+local function player_move(event)
+    local player = game.players[event.player_index]
+    if player.surface.name == variables.level.surface then  --check if the player has not been tped away
+        local center = centers[player.name]
+        if center then
+            local pos = player.position
+            local area = islands[player.name]
+            if not insideBox(area,pos) then
+                player.teleport({center.x,center.y},variables.level.surface)
+            end
+        end
+    end
+end
+
+local function on_player_left_game(event)
+    local player = game.players[event.player_index]
+
+    local inv = player.get_inventory(defines.inventory.god_main)
+    local all_items = inv.get_contents()
+    left_players[player.name] = all_items
+
+    player.set_controller {type = defines.controllers.god}
+    player.create_character()
+    player.teleport({-35,55},"nauvis")
+    --gui
+    local Main_gui = Gui.get_left_element(player, game_gui)
+    local table = Main_gui.container["slider"].table
+    table.visible = false
+    Gui.toggle_left_element(player,  game_gui, false)
+end
+
+local function player_join(event)
+    local player = game.players[event.player_index]
+    player.character.destroy()
+    local items = left_players[player.name]
+    local center = centers[player.name]
+    if items then
+        player.teleport(center,variables.level.surface)
+        if started[1] then
+            player.set_controller {type = defines.controllers.spectator}
+        else
+            player.set_controller {type = defines.controllers.god}
+            Gui.toggle_left_element(player, game_gui, true)
+            for item,amount in pairs(items) do
+                player.insert {name = item, count = amount}
+            end
+        end
+    else
+        player.set_controller {type = defines.controllers.spectator}
+        player.teleport(variables.level.center,variables.level.surface)
+    end
+
+
+end
+
+--gui
+local speed_slider
+local  function speed_change(_, element, _)
+    if Mini_games.get_running_game() == "Tight_spot" then
+        game.speed = element.slider_value
+    end
+end
+
+
+--game gui
+
+speed_slider =
+Gui.element{
+    type = "slider",
+    minimum_value = 1,
+    maximum_value  = 64,
+    value = 4,
+    value_step = 1,
+}:on_value_changed(speed_change)
+local label_func =
+Gui.element(function(_,parent,name,caption1,caption2)
+    parent.add {
+        type = "label",
+        caption = caption1,
+        style = "label",
```

Comment:
`label` is the default style for a label and so does not need to be defined here.